File: backend/src/main/java/com/policysquare/commercial/model/UnderwritingTip.java
package com.policysquare.commercial.model;

import jakarta.persistence.*;
import lombok.Data;
import java.time.LocalDateTime;

@Entity
@Data
@Table(name = "underwriting_tips")
public class UnderwritingTip {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private String id;

    private String title;
    
    private String category; // e.g. Fire Insurance, Marine, etc.
    
    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(columnDefinition = "TEXT")
    private String example;

    @Column(columnDefinition = "TEXT")
    private String keyTakeaway;

    private String imagePath; // Path to uploaded image

    @Column(updatable = false)
    private LocalDateTime createdAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
    }
}

File: policysquare/lib/data/models/underwriting_tip.dart
import 'package:json_annotation/json_annotation.dart';

part 'underwriting_tip.g.dart';

@JsonSerializable()
class UnderwritingTip {
  String? id;
  String title;
  String category;
  String? description;
  String? example;
  String? keyTakeaway;
  String? imagePath;
  String? createdAt;

  UnderwritingTip({
    this.id,
    required this.title,
    required this.category,
    this.description,
    this.example,
    this.keyTakeaway,
    this.imagePath,
    this.createdAt,
  });

  factory UnderwritingTip.fromJson(Map<String, dynamic> json) =>
      _$UnderwritingTipFromJson(json);
  Map<String, dynamic> toJson() => _$UnderwritingTipToJson(this);
}
